nautobot/core/templates/inc/nav_menu.html

```diff
@@ -2,27 +2,57 @@
                 <img src="{% custom_branding_or_static 'logo' 'img/nautobot_logo.svg' %}" height="30" />
             </a>
         </div>
 
-        <div id="navbar" class="navbar-collapse collapse">
+        <div id="navbar">
+            {% if request.user.is_authenticated or not "HIDE_RESTRICTED_UI"|settings_or_config %}
+            <form action="{% url 'search' %}" method="get" class="navbar-form" id="navbar_search" role="search">
+                <div class="input-group">
+                    <input type="text" name="q" class="form-control" placeholder="Search {{ settings.BRANDING_TITLE }}">
+                    <span class="input-group-btn">
+                        <button type="submit" class="btn btn-primary">
+                            <i class="mdi mdi-magnify"></i>
+                        </button>
+                    </span>
+                </div>
+            </form>
+            {% endif %}
+
+            <ul class="nav navbar-nav">
+                {% if request.user.is_authenticated %}
+                <li class="dropdown">
+                    <a href="#" class="dropdown-toggle" data-toggle="dropdown" title="{{ request.user }}" role="button" aria-haspopup="true" aria-expanded="false">
+                        <i class="mdi mdi-account"></i>
+                        <span id="navbar_user">{{ request.user|truncatechars:"30" }}</span>
+                        <span class="caret"></span>
+                    </a>
+                    <ul class="dropdown-menu">
+                        <li><a href="{% url 'user:profile' %}"><i class="mdi mdi-account-box"></i> Profile</a></li>
+                        {% if request.user.is_staff %}
+                        <li><a href="{% url 'admin:index' %}"><i class="mdi mdi-cogs"></i> Admin</a></li>
+                        {% endif %}
+                        <li class="divider"></li>
+                        <li><a href="{% url 'logout' %}"><i class="mdi mdi-logout"></i> Log out</a></li>
+                    </ul>
+                </li>
+                {% else %}
+                <li><a href="{% url settings.LOGIN_URL %}?next={{ request.get_full_path | urlencode }}"><i class="mdi mdi-login"></i> Log in</a></li>
+                {% endif %}
+            </ul>
+
             {% if request.user.is_authenticated %}
                 <ul class="nav navbar-nav">
                     {% for tab_name, tab_details in registry.nav_menu.tabs.items %}
                         {% if request.user|has_one_or_more_perms:tab_details.permissions or not "HIDE_RESTRICTED_UI"|settings_or_config %}
                             <li class="dropdown">
                                 <a href="#" class="dropdown-toggle" data-tab-weight="{{ tab_details.weight }}" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
+                                    <img src="{% custom_branding_or_static 'brand_chevron' 'img/nautobot_chevron.svg' %}" />
```

Comment:
Would be good to get input from @lampwins on whether we need this to be customizable branding or whether it should always just use the chevron. If the former, there needs to be some updates to the config template and the documentation to indicate this option; if the latter, this can probably just change to:

```suggestion
                                    <img src="{% static 'img/nautobot_chevron.svg' %}" />
```

Reply:
Sounds like we do want this to be customizable so this is appropriate - I'll take a pass later today at the config and docs to support this.